src/expr/skolem_manager.cpp

```diff
@@ -343,6 +349,24 @@ Node SkolemManager::getOriginalForm(Node n)
       {
         visited[cur] = cur.getAttribute(ofa);
       }
+      else if (cur.hasAttribute(ufa))
+      {
+        // if it has an unpurified form, compute the original form of it
+        Node ucur = cur.getAttribute(ufa);
+        if (ucur.hasAttribute(ofa))
+        {
+          // already computed, set
+          Node ucuro = ucur.getAttribute(ofa);
+          cur.setAttribute(ofa, ucuro);
+          visited[cur] = ucuro;
```

Comment:
this scenario seems strange. Could you clarify how the unpurified form could have had the original form set up but not the witness form?

Reply:
This always happens when we visit `cur` again, see "visit ucur then visit cur again" below.